src/cpp/core/CrashHandler.cpp

```diff
@@ -377,6 +384,16 @@ Error setUserHandlerEnabled(bool handlerEnabled)
    return Success();
 }
 
+bool hasUserBeenPromptedForPermission()
+{
+   return permissionFile().exists();
+}
+
+Error setUserHasBeenPromptedForPermission()
+{
+   return permissionFile().ensureFile();
```

Comment:
What would the behavior be if RStudio is unable to write this file for some reason (e.g. lacking permissions)? Would they get prompted every time RStudio starts up?

Reply:
Yes that is the behavior that would occur. The thinking here is that this should never happen in a valid deployment, since this is the user's personal settings directory.

Reply:
True, but for better or worse this does happen (e.g. somehow users end up with `~/.rstudio-desktop`, or parts of it, owned by `root` and so their user account can no longer read or write to that directory). RStudio then starts up in its default configuration so users are to some extent unaware (although other features eventually break down)

Perhaps we could also try writing something to the user's temporary directory as a fallback, and check whether either of those files exist?

Reply:
Does falling back to the temporary directory really buy us much? Considering the temp directory is often blown away after a restart of the machine, it seems to me it would slightly delay the inevitable.

Also this will be written under `~/.r` as opposed to `~/.rstudio-desktop` - not sure if that's a bit more resilient from the possible chown to root.

Reply:
How about detecting whether `~/.r` is writable, and treating a non-writeable `~/.r` as though the user has been prompted?